### PR TITLE
Set line length to 120 characters

### PR DESCRIPTION
**Pull request by Borda**
cc: @EnricoMi

### CODE REVIEW

**Comment by EnricoMi** [comment]
@JLLeitschuh @trim21 what do you think?

**Comment by trim21** [comment]
I suggest add a config file for black instead of hard-coded argument in pre-commit

**Comment by Borda** [comment]
> I suggest add a config file for black instead of hard-coded argument in pre-commit

Added `pyproject.toml` which we can also use for #2617

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+[tool.black]
+# https://github.com/psf/black
+line-length = 120
+exclude = "(.eggs|.git|.hg|.mypy_cache|.venv|_build|buck-out|build|dist)"
```
Comment (trim21) [inline pyproject.toml:4]: I think the default exclude already cover this, isn't it?

Comment (Borda) [reply pyproject.toml:4]:
```suggestion
```

**Comment by trim21** [comment]
maybe also add `target-version = ['py37']`

**Comment by Borda** [comment]
> target-version = ['py37']

does not have any effect, but sdded

**Comment by trim21** [comment]
> > target-version = ['py37']
> 
> does not have any effect, but sdded

looks like it does 

https://github.com/PyGithub/PyGithub/pull/2599/commits/0e49ad8b6c5b1a39f4c22e60723c3f35dbdda9cf#diff-d2a693bccbc5953aa099d38d5c73594e3cec275e229b18c820f80b43a64a51c0

**Review by trim21 — APPROVED** [review]
LGTM

File: pyproject.toml [hunk 2]
```diff
@@ -1,4 +1,3 @@
 [tool.black]
 # https://github.com/psf/black
 line-length = 120
-target-version = ['py37']
```
Comment (trim21) [inline pyproject.toml]: why removed?

**Comment by Borda** [comment]
> > setting 3.7 in #2594
> 
> no, it's not the same thing. tools like black and ruff doesn't infer target-version from the python version it's running.

reverted

**Comment by trim21** [comment]
lgtm

**Comment by trim21** [comment]
one more thing: 

https://github.com/PyGithub/PyGithub/blob/61f9ae6c4b943124dca93c14de54506a7661d68e/.isort.cfg#L6

**Review by JLLeitschuh — APPROVED** [review]
LGTM!

**Review by EnricoMi — APPROVED** [review]
LGTM!

**Comment by EnricoMi** [comment]
> one more thing:
> 
> https://github.com/PyGithub/PyGithub/blob/61f9ae6c4b943124dca93c14de54506a7661d68e/.isort.cfg#L6

@Borda what about this line length setting?

**Comment by Borda** [comment]
> @Borda what about this line length setting?

thx for reminder, yes, just set in last commit to be 120 too

**Comment by EnricoMi** [comment]
Awesome, thanks!

**Pull request merged**